### ISSUE TITLE
running aspnetcore app to azure app service suddenly return 502.5

### ISSUE THREAD

**Issue by TechHome-fuyanwen** [aspnet/IISIntegration#392]
Development environment:
dotnet version 1.0.0-preview2-003131
VS2015

A few days ago my application was normal，suddenly，service return HTTP Error 502.5 . I re-built an app service in azure, and deploy  the default VS2015 webapp template, but the error still exists。

I use KUDU to view the log, the error is as follows:
In Eventlog.xml:
      `Application 'MACHINE/WEBROOT/APPHOST/TECLINK' with physical root 'D:\home\site\wwwroot\' 
failed to start process with commandline 'dotnet .\*.dll', ErrorCode = '0x80004005 : e0434352\`
In stdout log:
     `warn: Microsoft.AspNetCore.Server.Kestrel[0]Unable to bind to http://localhost:5000 on the IPv4 loopback interface.
System.AggregateException: One or more errors occurred. (Error -4092 EACCES permission denied) 
 ---> Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.UvException: Error -4092 EACCES permission denied
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.Libuv.Check(Int32 statusCode)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.Libuv.tcp_bind(UvTcpHandle handle, SockAddr& addr, Int32 flags)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.UvTcpHandle.Bind(ServerAddress address)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.TcpListener.CreateListenSocket()
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.Listener.<>c.<StartAsync>b__6_0(Object state)
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Microsoft.AspNetCore.Server.Kestrel.Internal.KestrelEngine.CreateServer(ServerAddress address)
   at Microsoft.AspNetCore.Server.Kestrel.KestrelServer.Start[TContext](IHttpApplication`1 application)
---> (Inner Exception #0) Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.UvException: Error -4092 EACCES permission denied
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.Libuv.Check(Int32 statusCode)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.Libuv.tcp_bind(UvTcpHandle handle, SockAddr& addr, Int32 flags)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.UvTcpHandle.Bind(ServerAddress address)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.TcpListener.CreateListenSocket()
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.Listener.<>c.<StartAsync>b__6_0(Object state)<---

warn: Microsoft.AspNetCore.Server.Kestrel[0]
      Unable to bind to http://localhost:5000 on the IPv6 loopback interface.
System.AggregateException: One or more errors occurred. (Error -4090 EADDRNOTAVAIL address not available) ---> 
	  Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.UvException: Error -4090 EADDRNOTAVAIL address not available
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.Libuv.Check(Int32 statusCode)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.Libuv.tcp_bind(UvTcpHandle handle, SockAddr& addr, Int32 flags)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.UvTcpHandle.Bind(ServerAddress address)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.TcpListener.CreateListenSocket()
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.Listener.<>c.<StartAsync>b__6_0(Object state)
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Microsoft.AspNetCore.Server.Kestrel.Internal.KestrelEngine.CreateServer(ServerAddress address)
   at Microsoft.AspNetCore.Server.Kestrel.KestrelServer.Start[TContext](IHttpApplication`1 application)
---> (Inner Exception #0) Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.UvException: Error -4090 EADDRNOTAVAIL address not available
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.Libuv.Check(Int32 statusCode)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.Libuv.tcp_bind(UvTcpHandle handle, SockAddr& addr, Int32 flags)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.UvTcpHandle.Bind(ServerAddress address)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.TcpListener.CreateListenSocket()
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.Listener.<>c.<StartAsync>b__6_0(Object state)<---

fail: Microsoft.AspNetCore.Server.Kestrel[0]
      Unable to bind to http://localhost:5000 on any loopback interface.
System.AggregateException: One or more errors occurred. (Error -4092 EACCES permission denied) (Error -4090 EADDRNOTAVAIL address not available) ---> Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.UvException: Error -4092 EACCES permission denied
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.Libuv.Check(Int32 statusCode)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.Libuv.tcp_bind(UvTcpHandle handle, SockAddr& addr, Int32 flags)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.UvTcpHandle.Bind(ServerAddress address)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.TcpListener.CreateListenSocket()
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.Listener.<>c.<StartAsync>b__6_0(Object state)
   --- End of inner exception stack trace ---
---> (Inner Exception #0) Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.UvException: Error -4092 EACCES permission denied
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.Libuv.Check(Int32 statusCode)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.Libuv.tcp_bind(UvTcpHandle handle, SockAddr& addr, Int32 flags)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.UvTcpHandle.Bind(ServerAddress address)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.TcpListener.CreateListenSocket()
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.Listener.<>c.<StartAsync>b__6_0(Object state)<---

---> (Inner Exception #1) Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.UvException: Error -4090 EADDRNOTAVAIL address not available
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.Libuv.Check(Int32 statusCode)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.Libuv.tcp_bind(UvTcpHandle handle, SockAddr& addr, Int32 flags)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.UvTcpHandle.Bind(ServerAddress address)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.TcpListener.CreateListenSocket()
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.Listener.<>c.<StartAsync>b__6_0(Object state)<---`
What is causing the error? Thank you.

**Comment by Tratcher** [comment]
It shouldn't be trying to bind to http://localhost:5000 in azure/IIS.

Why are you still using 1.0 previews? We'll need you to repro this on released 1.0 or 1.1 bits before we can investigate.

**Comment by TechHome-fuyanwen** [comment]
@Tratcher hi, i use  the default VS2017 webapp template and dotnet sdk version is 1.1, but the problem still exists; then i try deploy the webapp to other azure account ,it's work ok!

**Comment by Tratcher** [comment]
Are the logs much different?

**Comment by TechHome-fuyanwen** [comment]
@Tratcher there are some different place,;
the normal one : 
`warn: Microsoft.AspNetCore.Server.Kestrel[0]
      Unable to bind to http://localhost:30464 on the IPv6 loopback interface: (Error -4090 EADDRNOTAVAIL address not available)
Hosting environment: Production
Content root path: D:\home\site\wwwroot
Now listening on: http://localhost:30464
Application started. Press Ctrl+C to shut down.`

the another is : 
    `warn: Microsoft.AspNetCore.Server.Kestrel[0]
      Unable to bind to http://localhost:29330 on the IPv4 loopback interface: (Error -4092 EACCES permission denied)
warn: Microsoft.AspNetCore.Server.Kestrel[0]
      Unable to bind to http://localhost:29330 on the IPv6 loopback interface: (Error -4090 EADDRNOTAVAIL address not available)
crit: Microsoft.AspNetCore.Server.Kestrel[0]
      Unable to start Kestrel.
System.IO.IOException: Failed to bind to address http://localhost:29330. ---> System.AggregateException: One or more errors occurred. (Error -4092 EACCES permission denied) (Error -4090 EADDRNOTAVAIL address not available) ---> Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.UvException: Error -4092 EACCES permission denied
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.Libuv.tcp_bind(UvTcpHandle handle, SockAddr& addr, Int32 flags)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.UvTcpHandle.Bind(ServerAddress address)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.TcpListener.CreateListenSocket()
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.Listener.<StartAsync>b__8_0(Object state)
   --- End of inner exception stack trace ---
   --- End of inner exception stack trace ---
   at Microsoft.AspNetCore.Server.Kestrel.KestrelServer.Start[TContext](IHttpApplication`1 application)

Unhandled Exception: System.IO.IOException: Failed to bind to address http://localhost:29330. ---> System.AggregateException: One or more errors occurred. (Error -4092 EACCES permission denied) (Error -4090 EADDRNOTAVAIL address not available) ---> Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.UvException: Error -4092 EACCES permission denied
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.Libuv.tcp_bind(UvTcpHandle handle, SockAddr& addr, Int32 flags)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.UvTcpHandle.Bind(ServerAddress address)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.TcpListener.CreateListenSocket()
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.Listener.<StartAsync>b__8_0(Object state)
   --- End of inner exception stack trace ---
   --- End of inner exception stack trace ---
   at Microsoft.AspNetCore.Server.Kestrel.KestrelServer.Start[TContext](IHttpApplication`1 application)
   at Microsoft.AspNetCore.Hosting.Internal.WebHost.Start()
   at Microsoft.AspNetCore.Hosting.WebHostExtensions.Run(IWebHost host, CancellationToken token, String shutdownMessage)
   at Microsoft.AspNetCore.Hosting.WebHostExtensions.Run(IWebHost host)
   at dynavinteclink.Program.Main(String[] args) in E:\Work\web\dynavinteclink\dynavinteclink\Program.cs:line 26`

**Comment by Tratcher** [comment]
@halter73 port conflicts?

**Comment by halter73** [comment]
The -4090 EADDRNOTAVAIL IPv6 failure is expected for app services.

Normally in the case of port conflicts, we would expect an -4091 EADDRINUSE, but in the case the port has already been reserved for something else by netsh or similar, we have seen -4092 EACESS [in the past](https://github.com/aspnet/KestrelHttpServer/issues/1265).

**Comment by TechHome-fuyanwen** [comment]
last day , i buy a Technical Support from azure , he said : the version on which my application is running is 2.0 preview version, which is only supported over github/forums. my issue is seems to be because of runtime version of the application, but they can not downgrade to older version. so they could not troubleshoot the issue further.

**Comment by dallancarr** [comment]
That command line seems a little suspect
``` 
dotnet .\*.dll
```

**Comment by TechHome-fuyanwen** [comment]
I changed my app service, and it's worked fine!

**Comment by JD-Robbs** [comment]
@TechHome-fuyanwen Can you provide some more details, please? I am facing the same issues.

**Comment by alex-erygin** [comment]
Hot works for me

**Comment by JesseXia** [comment]
I'm having the same exception here. Can you give more details, please?

**Comment by Tratcher** [comment]
@JesseXia please open a new issue with your details.
